tests/test_virtualenv.py

```diff
@@ -279,6 +279,66 @@ def test__resolved_interpreter_windows_pyexe_fails(sysfind, make_one):
     sysfind.assert_any_call("py")
 
 
+@pytest.mark.parametrize(
+    ["input_", "path_result", "expected"],
+    [
+        ("3.7", r"c:\python37-x64\python.exe", r"c:\python37-x64\python.exe"),
+        ("3.7", r"c:\fails\python.exe", None),
+        ("python3.7", r"c:\python37-x64\python.exe", None),
+        ("2.7", r"c:\python37-x64\python.exe", None),
+        ("goofy", r"c:\python37-x64\python.exe", None),
+        ("3.7", None, None),
+        ("python3.7", None, None),
+        ("2.7", None, None),
+        ("goofy", None, None),
+    ],
+)
+@mock.patch("nox.virtualenv._SYSTEM", new="Windows")
+@mock.patch.object(py.path.local, "sysfind")
+def test__resolved_interpreter_windows_path_and_version(
+    sysfind, make_one, input_, path_result, expected
+):
+    # Establish that if we get a standard pythonX.Y path, we look it
+    # up via the path on Windows.
+    venv, _ = make_one(interpreter=input_)
+
+    # Trick the system into thinking that it cannot find python3.6
```

Comment:
This comment refers to 3.6 but none of the test cases do. Then the next comment is about Unix but this test is meant to be Windows only?

Reply:
Thanks for taking the time to review! I was copying some logic from another test and did not fully adapt the comment, will change it.

One difficulty with testing `_resolved_parameter` is that it checks for things one after another, so every test has to make sure that the checks above the check-under-test fail (of which most are for Unix, hence the original comment). I am inclined to think that refactoring `_resolved_parameter` (separately from this PR, maybe next after) would help reducing the complexity of the tests (factoring out the checks themselves, separating them from the flow).

Reply:
SGTM